src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DataContext.hpp
+ */
+
+#ifndef GEOS_DATAREPOSITORY_SOURCECONTEXT_HPP_
+#define GEOS_DATAREPOSITORY_SOURCECONTEXT_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Logger.hpp"
+#include "xmlWrapper.hpp"
+
+namespace geos
+{
+
+namespace dataRepository
+{
+
+class Group;
+class WrapperBase;
+
+/// This abstract class stores data that helps to retrieve an object:
+/// - from which position in a file (if applicable), see DataFileContext,
+/// - where it is located in the data hierarchy, see GroupContext and WrapperContext.
+/// Typically, the target object contain an unique_ptr< DataContext > instance of this class.
```

Comment:
/// DataContext is an abstract class to storing contextual information on an object:
/// - its line position in a file if applicable, implementation in DataFileContext,
/// - its location in the data hierarchy, implementation in GroupContext and WrapperContext.
/// The target object contains a unique_ptr< DataContext > instance.